libs/domain/site/navigation-button/src/navigation-button.component.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { fixture } from '@open-wc/testing-helpers';
+import { useComponent } from '@spryker-oryx/core/utilities';
+import { html } from 'lit';
+import { beforeEach } from 'vitest';
+import { NavigationButtonComponent } from './navigation-button.component';
+import { navigationButtonComponent } from './navigation-button.def';
+
+describe('NavigationButtonComponent', () => {
+  let element: NavigationButtonComponent;
+
+  beforeAll(async () => {
+    await useComponent(navigationButtonComponent);
+  });
+
+  beforeEach(async () => {
+    element = await fixture(html`
+      <oryx-site-navigation-button text="test"></oryx-site-navigation-button>
+    `);
+  });
+
+  it('passes the a11y audit', async () => {
+    await expect(element).shadowDom.to.be.accessible();
+  });
+
+  it('should render oryx-heading', () => {
+    expect(element).toContainElement('oryx-heading');
+  });
+
+  it('should render button', () => {
+    expect(element).toContainElement('button');
+  });
+
+  describe('when text is not provided', () => {
+    beforeEach(async () => {
+      element = await fixture(html`
+        <oryx-site-navigation-button></oryx-site-navigation-button>
+      `);
+    });
+
+    it('should not render oryx-heading', () => {
+      expect(element).not.toContainElement('oryx-heading');
+    });
+  });
```

Comment:
It is double work, not sure that this is necessary. We have a conditional rendering in the code and it's enough just to test the condition when it should render something. Otherwise means that it should not

Reply:
We should be sure that it works correctly. Somebody can change the conditions and it will render when we are not expecting it. Unit tests should catch it before somebody will mention it on CR or even on production :D

Reply:
but again, if someone makes some changes then he|she should align unit-tests. We should not cover future cases. And the same, conditional rendering here is area of responsibility  of `when` directive. And for falsy condition we even do not have separate render function. And if we test this falsy result -> we will test behavior of `when` that is out of our current testing behavior

Reply:
I'm not talking that we should have those checks in `when text is not provided` scope. But I think that it is good to test the component without any provided attribute. 
- when no attribute provided:
-- should not render `oryx-heading` component
-- should not render `oryx-icon` component
-- should not render `mark` tag 
-- ...
I think that we should test it to be sure that our conditions work correctly even if the false condition renders nothing, but we should check it.